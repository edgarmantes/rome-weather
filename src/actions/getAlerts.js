
function resolvedGetAlert(data){
	return {
		type: 'GET_ALERTS',
		payload: data
	}
}



export default function getAlerts(){

	const URL = "http://api.openweathermap.org/data/3.0/triggers?appid=698009a4dde62e48490582289f422bd2";

	return dispatch => {
		return fetch(URL).then(  
			response => response.json()		
		).then( (object) => {
			
			let query = object.filter(function(object){
					
				return object.conditions[0]._id === "5963fe7891f06a005b5834b2"
				
			})

			// let alertsList = query[0].filter( function(object){
			// 		console.log(object)
			// 	return object
			// })

			console.log("actions_alerts",query[0].alerts, object)
			return dispatch(resolvedGetAlert(object))
		}).catch( (error) => {
			alert(error)
			return console.log(error)
		})
	}

}