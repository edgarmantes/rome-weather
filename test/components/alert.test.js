import React from 'react';
import { Provider } from 'react-redux';
import AlertsContainer, { Alerts } from '../../src/components/alerts';
import renderer from 'react-test-renderer';

import getAlerts from '../../src/actions/getAlerts';

import configureStore from 'redux-mock-store';

describe('<Alerts />' , () => {
	const initialState ={

		getAlerts:{
			alerts : [
				{
	  				"date": ["Mon", "Dec", "19", "2016", "15:00:00", "GMT-0600", "(Central", "Standard", "Time)"], 
	  				"hi": "-9"
	  			},
	  			{
	  				"date": ["Mon", "Dec", "19", "2016", "15:00:00", "GMT-0600", "(Central", "Standard", "Time)"], 
	  				"hi": "-9"
	  			},
	  			{
	  				"date": ["Mon", "Dec", "19", "2016", "15:00:00", "GMT-0600", "(Central", "Standard", "Time)"], 
	  				"hi": "-9"
	  			},
	  			{
	  				"date": ["Mon", "Dec", "19", "2016", "15:00:00", "GMT-0600", "(Central", "Standard", "Time)"], 
	  				"hi": "-9"
	  			},
	  			{
	  				"date": ["Mon", "Dec", "19", "2016", "15:00:00", "GMT-0600", "(Central", "Standard", "Time)"], 
	  				"hi": "-9"
	  			}

			]
		}, 
		changeUnit : {
			unit : {
				props : {
					children : "F"
				}
			}
		},
		currentTemp : { unit : "80" } 
	}
	const mockStore = configureStore();
	let store,wrapper

    beforeEach(()=>{
        store = mockStore(initialState)
        wrapper = shallow( <AlertsContainer store={store} /> )

    })

    it('Render the connected(SMART) component', () => {

       expect(wrapper.length).toEqual(1)
    });

	it('Renders Alerts component properly', () => {
		shallow(<Alerts />)
	});

	it("Renders with correct state.getAlerts.alerts value", () => {
		expect(wrapper.node.props.store.getState().getAlerts.alerts).toBe(initialState.getAlerts.alerts)
	})

	it("Renders with correct state.changeUnit.unit.props.children value", () => {
		expect(wrapper.node.props.store.getState().changeUnit.unit.props.children).toBe(initialState.changeUnit.unit.props.children)
	})

	it("Renders with correct state.currentTemp.unit value", () => {
		expect(wrapper.node.props.store.getState().currentTemp.unit).toBe(initialState.currentTemp.unit)
	})

	it('Snapshot of Alerts', () => {
		const wrapper = ''
        const renderedValue =  renderer.create(<Alerts dispatch={jest.fn()} />).toJSON()
        expect(renderedValue).toMatchSnapshot();
    }); 

});
