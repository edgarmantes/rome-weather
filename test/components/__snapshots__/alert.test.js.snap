// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alerts /> Snapshot of Alerts 1`] = `
<div
  className="alerts-container col-6"
>
  <h1
    className="alerts-hdr"
  >
    Days over 70℉ / 21℃
  </h1>
  <ul />
</div>
`;
